### PR TITLE
Dont let the pesky (0,0) jump after the go to definition action.

### PR DESCRIPTION
**Pull request by magniff**
**What this PR does / why we need it**:
"Fixing" an extremely annoying bug.

**Which issue(s) this PR fixes**
Addressing the Annoying Cursor Bug Affecting Go to Definition Actions.
You can find a ton of already existing chats, mentioning that specific bug, e.g. https://github.com/VSCodeVim/Vim/issues/8200

**Special notes for your reviewer**:
This is a hack that seems to be working, but, we need a better solution for sure.

**Details:**
The bug is hard to reproduce on purpose, it seem to just appear after 5-10 minutes of intense using of the plugin. At some point, after an yet another jump action (usually after the jump to definition), when you jump back by pressing `ctrl+o` [here](https://github.com/VSCodeVim/Vim/blob/master/src/mode/modeHandler.ts#L137) `vscode` returns a single `([0, 0] | [0, 0])` selection, which causes the mess. Have no idea why, maybe the bug of the `vscode` itself?

### CODE REVIEW

**Comment by magniff** [comment]
@Chillee or someone else, review please. The bug in question is making the user experience unbearable. Thanks!)

**Comment by FrankFang** [comment]
Why not merge the pr?

**Comment by Gredelston** [comment]
cc @J-Fields, can you please review this PR? This fixes a bug that makes the Vim extension quite unpleasant to use.

File: src/mode/modeHandler.ts [hunk 1]
```diff
@@ -135,6 +136,13 @@ export class ModeHandler implements vscode.Disposable, IModeHandler {
   public syncCursors() {
     // TODO: getCursorsAfterSync() is basically this, but stupider
     const { selections } = this.vimState.editor;
+    if (
+      selections.length == 1 &&
```
Comment (J-Fields) [inline src/mode/modeHandler.ts:140]: Needs to be `===`

File: src/mode/modeHandler.ts [hunk 2]
```diff
@@ -135,6 +136,13 @@ export class ModeHandler implements vscode.Disposable, IModeHandler {
...
+      selections[0].isEqual(new Range(new Position(0, 0), new Position(0, 0)))
+    ) {
+      return;
```
Comment (J-Fields) [inline src/mode/modeHandler.ts:143]: Let's put a `TODO` here referencing this MR

**Review by J-Fields — APPROVED** [review]
Thanks!

**Pull request merged**